src/signs_cmd.cpp

```diff
@@ -42,16 +42,11 @@ std::tuple<CommandCost, SignID> CmdPlaceSign(DoCommandFlags flags, TileIndex til
 
 	/* When we execute, really make the sign */
 	if (flags.Test(DoCommandFlag::Execute)) {
-		Sign *si = new Sign(_game_mode == GM_EDITOR ? OWNER_DEITY : _current_company);
 		int x = TileX(tile) * TILE_SIZE;
 		int y = TileY(tile) * TILE_SIZE;
 
-		si->x = x;
-		si->y = y;
-		si->z = GetSlopePixelZ(x, y);
-		if (!text.empty()) {
-			si->name = text;
-		}
+		Sign *si = new Sign(_game_mode == GM_EDITOR ? OWNER_DEITY : _current_company, x, y, GetSlopePixelZ(x, y), text);
```

Comment:
This no longer checks if `!text.empty()`. Not sure if that changes anything.

Reply:
It's a potential micro-optimisation you'd not be doing now... or not. It depends a bit on what the constructor does exactly.

You could claim that with the current code the constructor would first construct an empty string, thus zero-ing specific memory. Then it'd be overwriting it with a new state when the sign has a text.

With the new code the constructor can avoid setting some fields to zero when it doesn't need to.

Functionally nothing changes: in the end `si->name` equals `name` when calling `si->UpdateVirtCoord()`.